README.md

```diff
@@ -1,2 +1,124 @@
 # big-json
-A stream based implementation of JSON.parse and JSON.stringify for big POJOs
+
+[![NPM Version](https://img.shields.io/npm/v/big-json.svg)](https://npmjs.org/package/big-json)
+[![Build Status](https://travis-ci.org/DonutEspresso/big-json.svg?branch=master)](https://travis-ci.org/DonutEspresso/big-json)
+[![Coverage Status](https://coveralls.io/repos/github/DonutEspresso/big-json/badge.svg?branch=master)](https://coveralls.io/github/DonutEspresso/big-json?branch=master)
+[![Dependency Status](https://david-dm.org/DonutEspresso/big-json.svg)](https://david-dm.org/DonutEspresso/big-json)
+[![devDependency Status](https://david-dm.org/DonutEspresso/big-json/dev-status.svg)](https://david-dm.org/DonutEspresso/big-json#info=devDependencies)
+[![bitHound Score](https://www.bithound.io/github/DonutEspresso/big-json/badges/score.svg)](https://www.bithound.io/github/DonutEspresso/big-json/master)
+[![nsp status](https://img.shields.io/badge/NSP%20status-no%20vulnerabilities-green.svg)](https://travis-ci.org/DonutEspresso/big-json)
+
+> A stream based implementation of JSON.parse and JSON.stringify for big POJOs
+
+There exist many stream based implementations of JSON parsing or stringifying
+for large data sets. These implementations typical target time series data, new
+line delimited data or other array-like data, e.g., logging records or other
+continuous flowing data.
+
+This module hopes to fill a gap in the ecosystem: parsing large JSON objects
+that are just _really_ big objects. With large in-memory objects, it is
+possible to run up against the V8 string length limitation, which is currently
+(as of 9/2017) 268435440 characters. Thus, if your large object has enough keys
+or values, it is possible to exceed the string length limit when calling
+[JSON.stringify](https://github.com/nodejs/node/issues/10738).
+
+Similarly, when retrieving stored JSON from disk or over the network, if the
```

Comment:
Might want to also talk about how this will also block the event loop, so streaming lets you chunk up JSON parsing so other tasks get a turn on CPU.